### PR TITLE
[Merged by Bors] - chore(RingTheory/Derivation/Basic): generalize `Derivation.compAlgebraMap` to semirings

### PR DESCRIPTION
**Pull request by eric-wieser**
Also adds a `simps` configuration so we get nice lemmas names with `@[simps]` and shuffles the variable order to put all the types first

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Derivation/Basic.lean [hunk 1]
```diff
@@ -40,8 +40,9 @@ equality. We also require that `D 1 = 0`. See `Derivation.mk'` for a constructor
 assumption from the Leibniz rule when `M` is cancellative.
 
 TODO: update this when bimodules are defined. -/
-structure Derivation (R : Type*) (A : Type*) [CommSemiring R] [CommSemiring A] [Algebra R A]
-    (M : Type*) [AddCommMonoid M] [Module A M] [Module R M] extends A →ₗ[R] M where
+structure Derivation (R : Type*) (A : Type*) (M : Type*)
```
Comment (riccardobrasca) [inline Mathlib/RingTheory/Derivation/Basic.lean:43]: I guess `A` can even be a `Semiring`, at least in this definition.

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/RingTheory/Derivation/Basic.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/RingTheory/Derivation/Basic.lean:43]: That wouldn't make any sense without https://github.com/leanprover-community/mathlib/pull/18936

Comment (riccardobrasca) [reply Mathlib/RingTheory/Derivation/Basic.lean:43]: Ah yes, `Module` still wants a commutative ring.

**Comment by riccardobrasca** [comment]
BTW, where is now "Note [custom simps projection]"? There is no hyperlink in the doc.

**Comment by eric-wieser** [comment]
Library notes are not yet implemented in doc-gen4 (or vscode) :(

File: Mathlib/RingTheory/Derivation/Basic.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/RingTheory/Derivation/Basic.lean:43]: That's not quite true; the generalization you suggest is accepted by Lean, but it is not a reasonable definition for the non-commutative case.

**Pull request closed without merging**